--- a/gpt_engineer/steps.py
+++ b/gpt_engineer/steps.py
@@ -18,21 +18,21 @@
         ai.fsystem(dbs.preprompts["use_feedback"]),
     ]
     messages = ai.next(messages, dbs.input["feedback"])
     to_files(messages[-1]["content"], dbs.workspace)
     return messages
 
 
 def fix_code(ai: AI, dbs: DBs):
-    code_ouput = json.loads(dbs.logs[gen_code.__name__])[-1]["content"]
+    code_output = json.loads(dbs.logs[gen_code.__name__])[-1]["content"]
     messages = [
         ai.fsystem(setup_sys_prompt(dbs)),
         ai.fuser(f"Instructions: {dbs.input['main_prompt']}"),
-        ai.fuser(code_ouput),
+        ai.fuser(code_output),
         ai.fsystem(dbs.preprompts["fix_code"]),
     ]
     messages = ai.next(messages, "Please fix any errors in the code above.")
     to_files(messages[-1]["content"], dbs.workspace)
     return messages
 
 
 class Config(str, Enum):
