--- a/scripts/benchmark.py
+++ b/scripts/benchmark.py
@@ -1,8 +1,9 @@
+                    bench_folder,
                     "--steps",
                     "benchmark",
                 ],
                 stdout=log_file,
                 stderr=log_file,
                 bufsize=0,
             )
             benchmarks.append((bench_folder, process, log_file))
@@ -17,25 +18,23 @@
 
         print("process", bench_folder.name, "finished with code", process.returncode)
         print("Running it. Original benchmark prompt:")
         print()
         with open(bench_folder / "main_prompt") as f:
             print(f.read())
         print()
 
-        try:
+        with contextlib.suppress(KeyboardInterrupt):
             subprocess.run(
                 [
                     "python",
                     "-m",
                     "gpt_engineer.main",
                     bench_folder,
                     "--steps",
                     "execute_only",
                 ],
             )
-        except KeyboardInterrupt:
-            pass
 
 
 if __name__ == "__main__":
     run(main)
