--- a/gpt_engineer/steps.py
+++ b/gpt_engineer/steps.py
@@ -1,36 +1,42 @@
             "a) install dependencies "
-            "b) run the necessary parts of the codebase to try it.\n"
+            "b) run all necessary parts of the codebase (in parallell if necessary).\n"
+            "Do not install globally. Do not use sudo.\n"
             "Do not explain the code, just give the commands.\n"
         ),
         user="Information about the codebase:\n\n" + dbs.workspace["all_output.txt"],
     )
 
     [[lang, command]] = parse_chat(messages[-1]['content'])
     assert lang in ['', 'bash', 'sh']
+
+    dbs.workspace['run.sh'] = command
 
     print('Do you want to execute this code?')
     print(command)
     print()
     print('If yes, press enter. If no, type "no"')
     print()
     if input() == 'no':
         print('Ok, not executing the code.')
         return messages
     print('Executing the code...')
     print()
-    subprocess.run(command, shell=True)
+
+    # Run the subprocess in dbs.workspace.path
+    subprocess.run('bash run.sh', shell=True, cwd=dbs.workspace.path)
     return messages
 
 
 # Different configs of what steps to run
 STEPS = {
     'default': [gen_spec, pre_unit_tests, run_clarified],
     'simple': [run, execute_workspace],
     'clarify': [clarify, run_clarified],
+    'execute_only': [execute_workspace],
 }
 
 # Future steps that can be added:
 # self_reflect_and_improve_files,
 # add_tests
 # run_tests_and_fix_files,
 # improve_based_on_in_file_feedback_comments
