--- a/gpt_engineer/db.py
+++ b/gpt_engineer/db.py
@@ -1,38 +1,43 @@
 from dataclasses import dataclass
-import os
 from pathlib import Path
 
 
+# This class represents a simple database that stores its data as files in a directory.
 class DB:
     """A simple key-value store, where keys are filenames and values are file contents."""
 
     def __init__(self, path):
         self.path = Path(path).absolute()
-        os.makedirs(self.path, exist_ok=True)
+
+        self.path.mkdir(parents=True, exist_ok=True)
+
+    def __contains__(self, key):
+        return (self.path / key).is_file()
 
     def __getitem__(self, key):
-        try:
-            with open(self.path / key, encoding='utf-8') as f:
-                return f.read()
-        except FileNotFoundError:
+        full_path = self.path / key
+
+        if not full_path.is_file():
             raise KeyError(key)
+        with full_path.open("r", encoding="utf-8") as f:
+            return f.read()
 
     def __setitem__(self, key, val):
-        Path(self.path / key).absolute().parent.mkdir(parents=True, exist_ok=True)
+        full_path = self.path / key
+        full_path.parent.mkdir(parents=True, exist_ok=True)
 
-        with open(self.path / key, 'w', encoding='utf-8') as f:
-            f.write(str(val))
-
-    def __contains__(self, key):
-        return (self.path / key).exists()
+        if isinstance(val, str):
+            full_path.write_text(val, encoding="utf-8")
+        else:
+            # If val is neither a string nor bytes, raise an error.
+            raise TypeError("val must be either a str or bytes")
 
 
+# dataclass for all dbs:
 @dataclass
 class DBs:
-    """A dataclass for all dbs"""
-
     memory: DB
     logs: DB
-    identity: DB
+    preprompts: DB
     input: DB
     workspace: DB
