--- a/gpt_engineer/main.py
+++ b/gpt_engineer/main.py
@@ -18,17 +18,15 @@
         workspace=DB(workspace_path),
         preprompts=DB(Path(__file__).parent / "preprompts"),
         archive=None if initial_run else DB(archive_path),
     )
 
     steps = STEPS[steps_config]
     for step in steps:
         messages = step(ai, dbs)
-        # TODO: Don't dump if step returns None
-        # human_review, archive don't need to be dumped
         dbs.logs[step.__name__] = json.dumps(messages)
 
     collect_learnings(model, temperature, steps, dbs)
 
 
 if __name__ == "__main__":
     app()
